### ISSUE TITLE
Card objects currently return both `brand` and `type`?

### ISSUE THREAD

**Issue by EddM** [rebelidealist/stripe-ruby-mock#168]
I guess this is, more generally, to start a discussion about backwards-compatibility vs. adherence to the Stripe API, but this issue I think is a good starting point.

Mocked card objects currently return both `brand` and the (defunct) `type` properties, so as to not break implementations relying on `type`.

Personally, I think when Stripe introduce a breaking change to their API, so should we, lest users write tests that pass with `stripe-ruby-mock`, but fail at the worst time when they update their live Stripe API version. Does everyone agree (especially @mindeavor)?

**Comment by sman591** [comment]
I actually implemented this back when Stripe changed their API. Looking back, I feel it would've been better to have 'stripe-ruby-mock' also break old implementations. Otherwise, like you mentioned, the change wouldn't be noticed until put to real use.

**Comment by mindeavor** [comment]
StripeMock should definitely keep up with the latest Stripe API version. If
StripeMock is up to date, it will help other developers upgrade their apps
to Stripe's latest API as well.

Every time we introduce a breaking change, we can reserve a branch for that
API version. I think this might be enough for those lagging behind in
versions. But that's just my guess.

**Comment by EddM** [comment]
Yeah I like that idea, keeping changes that apply to newer versions in different branches. (Maybe you want to set a target API version for a 2.0 release too?)

In bringing StripeMock up to speed (what little of it is left at this point) with its "base" version (`2014-06-17`), I assume we're cool introducing breaking changes to master as long as it matches the expected behaviour of 2014-06-17?

**Comment by mindeavor** [comment]
Yes on all accounts. Breaking changes can go in a api-yyyy-mm-dd
name formatted branch.

What did you mean by "set a target API version for a 2.0 release" ?

**Comment by mindeavor** [comment]
(back from my trip!) To be clear, what I meant was, anytime there is a breaking change, we create a new `api-yyyy-mm-dd` branch based on `master`. This new branch will be stable for that API version for those who need more time to update their code.

In other words, `master` will aim to always have the latest Stripe API behavior.

**Comment by EddM** [comment]
Sorry, totally forgot about this thread here.

Yeah I was suggesting something similar but that actually sounds even better and cleaner
